src/index.ts

```diff
@@ -1,64 +1,139 @@
 import { TypedError } from 'typed-error';
 
 export interface Credit {
+	validFrom: Date;
+	version: number;
 	firstDiscountPriceCents: number;
 	discountRate: number;
 	discountThreshold: number;
 	discountThresholdPriceCents: number;
 }
 
+interface Credits {
+	[slug: string]: Credit[];
+}
+
+interface Options {
+	credits?: Credits;
+	target?: 'current' | 'latest' | number | Date;
+}
+
 export class InvalidParametersError extends TypedError {
 	constructor(message: string) {
 		super(message);
 	}
 }
 
 // Credit pricing definitions.
-// Initially restrict credit usage to device:microservices feature.
-export const CREDITS: { [key: string]: Credit } = {
-	'device:microservices': {
-		firstDiscountPriceCents: 149,
-		discountRate: 0.33,
-		discountThreshold: 12000,
-		discountThresholdPriceCents: 125,
-	},
+const CREDITS: Credits = {
+	'device:microservices': [
+		{
+			version: 1,
+			validFrom: new Date('2023-02-01T00:00:00'),
```

Comment:
Should instead enforce UTC +0 with `new Date('2023-02-01T00:00:00.00+0000')`?

Reply:
Going with `new Date('2023-02-01T00:00:00Z')`, simpler and we only care about UTC+0.